compute_endpoint/globus_compute_endpoint/endpoint/endpoint.py

```diff
@@ -618,10 +617,10 @@ def stop_endpoint(
 
         log.debug(f"{ep_name} has a daemon.pid file")
 
-        if endpoint_config and endpoint_config.endpoint_teardown:
-            Endpoint._run_command(
-                "endpoint_teardown", endpoint_config.endpoint_teardown
-            )
+        if isinstance(endpoint_config, UserEndpointConfig):
+            teardown = endpoint_config and endpoint_config.endpoint_teardown
+            if teardown:
+                Endpoint._run_command("endpoint_teardown", teardown)
```

Comment:
It was a line length issue, and this was the cleanest approach I could come up with.

The suggested diff doesn't define `teardown`.    Putting `endpoint_config.endpoint_teardown` in it's places blows the line length and flake8 kicks in.

Reply:
My suggestion is to replace `endpoint_config and endpoint_config.endpoint_teardown` with just `endpoint_config.endpoint_teardown` because the `isinstance()` check already establishes that `endpoint_config` is truthy.

Reply:
Ah; that I can do.  But what's odd is I _did_ do exactly that at one point and mypy didn't like it.  (A fair chunk of time spent on this PR was satisfying mypy.)  No matter, something has shifted and now mypy is happy.  Good suggestion.